### PR TITLE
Fix the pkeyauth flow

### PR DESCRIPTION
**Pull request by heidijinxujia**
[passed] test case:[ Test Case 652091](https://identitydivision.visualstudio.com/IDDP/_workitems/edit/652091)

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/PKeyAuthChallengeHandler.java [hunk 1]
```diff
@@ -60,30 +60,35 @@ public PKeyAuthChallengeHandler(@NonNull final WebView view,
     public Void processChallenge(final PKeyAuthChallenge pKeyAuthChallenge) {
         mWebView.stopLoading();
         mChallengeCallback.setPKeyAuthStatus(true);
+        new Thread(new Runnable() {
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/PKeyAuthChallengeHandler.java]: Why a new Thread ? ` mWebView.post(new Runnable() { ` is again executed on UI thread anyways right ?

File: common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/PKeyAuthChallengeHandler.java [hunk 2]
```diff
@@ -105,8 +110,11 @@ public void run() {
                 if (privateKey == null) {
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/PKeyAuthChallengeHandler.java]: Wouldn't the if statement always be false ? This still needs to be fixed i believe , where we need to set the right thumbprint from the challenge headers and compare with the one in the device certificate

Comment (heidijinxujia) [reply common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/PKeyAuthChallengeHandler.java]: "thumbprint from the challenge headers" this only possibly happen when talking to :token endpoint.  This happens when broker receive device certificate challenge request during the token request. 

And currently, the code is missing this part. https://github.com/AzureAD/microsoft-authentication-library-common-for-android/issues/406 I'm going to fix this in other PR. This PR is focusing on fixing the pkeyauth challenge during the :auth flow.

**Comment by heidijinxujia** [comment]
#393 Fix

**Pull request merged**